pkg/cosign/keys.go

```diff
@@ -51,6 +60,73 @@ func GeneratePrivateKey() (*ecdsa.PrivateKey, error) {
 	return ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
 }
 
+func ImportKeyPair(keyPath string, pf PassFunc) (*Keys, error) {
+
+	kb, err := os.ReadFile(filepath.Clean(keyPath))
+	if err != nil {
+		return nil, err
+	}
+
+	p, _ := pem.Decode(kb)
+	if p == nil {
+		return nil, errors.New("invalid pem block")
+	}
+
+	if p.Type == RSAPrivateKeyPemType {
```

Comment:
maybe try a switch statement here?

also, `return MarshallKeyPair(Key{pk, pk.Public()}, pf)` could simply be at the end if the switch statement continues without hitting an unsupported line

Reply:
updated logic here.